src/main/java/leon/codegen/runtime/StrOps.java

```diff
@@ -0,0 +1,42 @@
+package leon.codegen.runtime;
+
+import org.apache.commons.lang3.StringEscapeUtils;
+
+public class StrOps {
+	public static String concat(String a, String b) {
+      return a + b;
+    }
+	public static BigInt length(String a) {
+      return new BigInt(String.valueOf(a.length()));
+    }
+	public static String substring (String a, BigInt start, BigInt end) {
+      if(start.greaterEquals(end) || start.greaterEquals(length(a)) || end.lessEquals(new BigInt("0")))
```

Comment:
Corrected in next commit. It came from the use of spaces and tabs which was invisible on my machine.